From 9d9af2b63b8f3f03191eeaeff3fd469a37c5373c Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Wed, 10 May 2017 14:19:03 -0500
Subject: [PATCH 12/16] Revert "TEMP: remoteproc/pru: enable ethernet fw for
 PRUSSs on K2G ICE EVM"

This reverts commit 8eabf4474a3c2300337450635a75d62700aeb689.

The PRU Ethernet driver will use the rproc_set_firmware() API to
configure the appropriate firmware names to be used by the remoteproc
core for loading and booting the PRU remoteprocs. Undo the changes
added previously in PRU remoteproc driver in commit 8eabf4474a3c
("TEMP: remoteproc/pru: enable ethernet fw for PRUSSs on K2G ICE EVM")
to perform the same.

NOTE:
The internal pinmux programming required for K2G-ICE is still in
place in the PRU remoteproc driver.

Signed-off-by: Suman Anna <s-anna@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 drivers/remoteproc/pru_rproc.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/remoteproc/pru_rproc.c b/drivers/remoteproc/pru_rproc.c
index f91a4da..3998160 100644
--- a/drivers/remoteproc/pru_rproc.c
+++ b/drivers/remoteproc/pru_rproc.c
@@ -731,8 +731,7 @@ static int pru_rproc_probe(struct platform_device *pdev)
 	if (of_machine_is_compatible("ti,am3359-icev2") ||
 	    of_machine_is_compatible("ti,am437x-idk-evm") ||
 	    of_machine_is_compatible("ti,am5728-idk") ||
-	    of_machine_is_compatible("ti,am5718-idk") ||
-	    of_machine_is_compatible("ti,k2g-ice")) {
+	    of_machine_is_compatible("ti,am5718-idk")) {
 		if (use_eth_fw && (pdata->caps & PRU_FUNC_CAPS_ETHERNET))
 			use_eth = true;
 	}
@@ -952,30 +951,22 @@ static int pru_rproc_remove(struct platform_device *pdev)
 /* 66AK2G PRUSS0 PRU core-specific private data */
 static struct pru_private_data k2g_pru0_0_rproc_pdata = {
 	.id = 0,
-	.caps = PRU_FUNC_CAPS_ETHERNET,
 	.fw_name = "k2g-pru0_0-fw",
-	.eth_fw_name = "ti-pruss/k2g-pru0-prueth-fw.elf"
 };
 
 static struct pru_private_data k2g_pru0_1_rproc_pdata = {
 	.id = 1,
-	.caps = PRU_FUNC_CAPS_ETHERNET,
 	.fw_name = "k2g-pru0_1-fw",
-	.eth_fw_name = "ti-pruss/k2g-pru1-prueth-fw.elf"
 };
 
 static struct pru_private_data k2g_pru1_0_rproc_pdata = {
 	.id = 0,
-	.caps = PRU_FUNC_CAPS_ETHERNET,
 	.fw_name = "k2g-pru1_0-fw",
-	.eth_fw_name = "ti-pruss/k2g-pru0-prueth-fw.elf"
 };
 
 static struct pru_private_data k2g_pru1_1_rproc_pdata = {
 	.id = 1,
-	.caps = PRU_FUNC_CAPS_ETHERNET,
 	.fw_name = "k2g-pru1_1-fw",
-	.eth_fw_name = "ti-pruss/k2g-pru1-prueth-fw.elf"
 };
 
 /* AM33xx SoC-specific PRU Device data */
-- 
1.9.1

